src/lib.rs

```diff
@@ -5,7 +5,7 @@ extern crate foxbox_taxonomy;
 extern crate transformable_channels;
 
 /// The back-end thread, in charge of the heavy lifting of managing adapters.
-mod backend;
+pub mod backend;
```

Comment:
Rather than bulk-reexporting, wouldn't it be possible to `pub use backend::WatchGuard` from the front-end?